client/my-sites/stats/stats-post-summary/index.jsx

```diff
@@ -74,18 +87,7 @@ class StatsPostSummary extends Component {
 					return [];
 				}
 
-				const months = flatten(
-					Object.keys( stats.years ).map( ( year ) => {
-						return range( 1, 13 ).map( ( month ) => {
-							const firstDayOfMonth = moment( `1/${ month }/${ year }`, 'DD/MM/YYYY' );
-							return {
-								period: firstDayOfMonth.format( 'MMM YYYY' ),
-								periodLabel: firstDayOfMonth.format( 'MMMM YYYY' ),
-								value: get( stats.years, [ year, 'months', month ], 0 ),
-							};
-						} );
-					} )
-				);
+				const months = [ ...statsByMonth( stats, moment ) ];
```

Comment:
That's an interesting use of generators. Very terse and easy to grok. Well done!

Reply:
I wanted to do a classic old-fashioned `for` loop to generate the sequence 1..12. And only with generators is that possible.